app/src/androidTestE2eTest/java/io/appium/uiautomator2/unittest/test/DeviceCommandsTest.java

```diff
@@ -469,26 +473,162 @@ public void toastVerificationTest4() throws JSONException {
     }
 
     /**
-     * Performs Scroll to specified element
+     * Performs scrolling through a horizontal list to an element specified by its text.
      *
      * @throws JSONException
      */
     @Test
-    public void scrollTest() throws JSONException {
-        scrollTo("Views"); // Due to 'Views' option not visible on small screen
-        Response response = findElement(By.accessibilityId("Views"));
-        clickAndWaitForStaleness(response.getElementId());
-        waitForElement(By.accessibilityId("Buttons"));
+    public void scrollHorizontalListTest() throws JSONException {
+        navigateTo("Views/Tabs/5. Scrollable");
 
-        String scrollToText = "WebView";
-        By by = By.accessibilityId(scrollToText);
-        response = findElement(by);
-        // Before Scroll 'Radio Group' Element was not found
-        assertFalse(by + " should not be found", response.isSuccessful());
-        scrollTo(scrollToText);
-        response = findElement(by);
-        // After Scroll Element was found
-        assertTrue(by + " should be found", response.isSuccessful());
+        // On devices with Android API Level 23 and earlier, the tab labels are displayed
+        // in all-caps, whereas their actual text is rather like 'Tab 13', 'Tab 26', etc.
+        // The arguments of methods findElement() and scrollToText() are case-sensitive,
+        // therefore we need to use different sets of inputs for different target devices.
+        int apiLevel = (int) getField(UiDevice.class, "API_LEVEL_ACTUAL", getUiDevice());
```

Comment:
I assume there are easier ways to get the actual api level

Reply:
https://github.com/appium/appium-uiautomator2-server/blob/20de6220d4d9b888892b8e148a1e16b6aa8b874d/app/src/main/java/io/appium/uiautomator2/utils/DeviceInfoHelper.java#L78

Reply:
Thank you for the suggestion. This method returns a string which might be not as easy to use as an integer. I'll see what could be done with that.

Reply:
The current implementation that uses integers seems to serve the purpose quite well. I would rather keep it this way.